src/manducky.py

```diff
@@ -18,72 +19,95 @@
-            self.templates["shirt"] = Image.open(ASSETS_PATH / "templates/variation_2/shirt.png")
-            self.templates["pants"] = Image.open(ASSETS_PATH / "templates/variation_2/pants.png")
+        if variation_ == 1:
+            template["dress"] = (
+                Image.open(ASSETS_PATH / "templates/variation_1/dress.png"),
+                dress_colors.shirt
+            )
+        else:
+            template['shirt'] = (
+                Image.open(ASSETS_PATH / "templates/variation_2/shirt.png"),
+                dress_colors.shirt
+            )
+            template['pants'] = (
+                Image.open(ASSETS_PATH / "templates/variation_2/pants.png"),
+                dress_colors.pants
+            )
 
         if ducky.hat:
-            self.templates["hat"] = Image.open(ASSETS_PATH / f"accessories/hats/{ducky.hat}.png")
+            template["hat"] = (
+                Image.open(ASSETS_PATH / f"accessories/hats/{ducky.hat}.png"),
+            )
         if ducky.outfit:
-            self.templates["outfit"] = Image.open(
-                ASSETS_PATH / f"outfits/variation_{self.variation}/{ducky.outfit}.png"
+            template["outfit"] = (
+                Image.open(ASSETS_PATH / f"outfits/{variation}/{ducky.outfit}.png"),
             )
         if ducky.equipment:
-            self.templates["equipment"] = Image.open(
-                ASSETS_PATH / f"equipment/variation_{self.variation}/{ducky.equipment}.png"
+            template["equipment"] = (
+                Image.open(ASSETS_PATH / f"equipment/{variation}/{ducky.equipment}.png"),
             )
 
-        self.hat = ducky.hat
-        self.equipment = ducky.equipment
-        self.outfit = ducky.outfit
-
-    def generate(self) -> ManDucky:
-        """Actually generate the ducky."""
-        dress_colors = make_man_duck_colors()
-
-        if self.variation == 2:
-            self.apply_layer(self.templates["pants"], dress_colors.pants)
-
-        self.apply_layer(self.templates["bill"], self.colors.beak)
-        self.apply_layer(self.templates["head"], self.colors.body)
-        self.apply_layer(self.templates["eye"], self.colors.eye)
-
-        if self.variation == 2:
-            self.apply_layer(self.templates["shirt"], dress_colors.shirt)
-
-        elif self.variation == 1:
-            self.apply_layer(self.templates["dress"], dress_colors.shirt)
-
-        if self.outfit and self.outfit != "bread":
-            self.apply_layer(self.templates["outfit"])
-
-        if self.equipment:
-            self.apply_layer(self.templates["equipment"])
-
-        self.apply_layer(self.templates["hands"], self.colors.wing)
-
-        if self.outfit and self.outfit == "beard":
-            self.apply_layer(self.templates["outfit"])
+        return template
+
+    def generate_from_options(self, options: Optional[ManDuckRequest]) -> dict:
+        """Generate a man duck from the provided request."""
+        colors = DuckyColors(**options["colors"])
```

Comment:
Two things to bear in mind here:
1. The objects derived from pydantic's BaseModel support dot access, so it would be better in all cases where you're subscripting the objects like this to just do it like `options.colors`.
2. Objects derived from pydantic's BaseModel are not mappings, and as such can't be used with `**`, instead to use them like this you can call `options.colors.dict()` on them.

Do note that this comment also applies to the other places in the file you do this on the next few lines too, which should also be fixed. After that's done I'll test again and we'll go from there.

Reply:
Thanks for the explanation, I will do the required changes, this is how it was done for a `Ducky Request` so I went ahead and used the same implementation